ee/secureenclaverunner/secureenclaverunner.go

```diff
@@ -0,0 +1,305 @@
+//go:build darwin
+// +build darwin
+
+package secureenclaverunner
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os/user"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/kolide/launcher/ee/agent/types"
+	"github.com/kolide/launcher/ee/consoleuser"
+	"github.com/kolide/launcher/pkg/backoff"
+	"github.com/kolide/launcher/pkg/traces"
+)
+
+const (
+	publicEccDataKey = "publicEccData"
+)
+
+type secureEnclaveRunner struct {
+	uidPubKeyMap        map[string]*ecdsa.PublicKey
+	secureEnclaveClient secureEnclaveClient
+	store               types.GetterSetterDeleter
+	slogger             *slog.Logger
+	mux                 *sync.Mutex
```

Comment:
I think it would be helpful to rename `mux` to make it clearer what resource (I think `uidPubKeyMap`) is being locked, or what kind of concurrent action we're trying to prevent

Reply:
done